### PR TITLE
Check for e.isFakeIEEvent has been deleted from page-unload-barrier (closes #608)

### LINKED ISSUE #608: Fix typo in the page-unload-barrier
https://github.com/DevExpress/testcafe/blob/9950f6bc253b07c0f8e8bed730a2d98f8f5dc74f/src/client/core/page-unload-barrier.js#L42

`e.isFakeIEBeforeUnloadEvent -> e.isFakeIEEvent`

### PR DESCRIPTION
**Pull request by georgiy-abbasov**
\cc @AlexanderMoskovkin, @inikulin

### CODE REVIEW

**Comment by AlexanderMoskovkin** [comment]
We should add a regression test for any fix if it's possible. Is there a reason why we can't add a test here?

**Comment by inikulin** [comment]
Why iframe test fails again?

**Comment by AlexanderMoskovkin** [comment]
> Why iframe test fails again?

Something went wrong on Firefox in Safari.
	
In a nutshell: we click on a link in an iframe (after that iframe is reloading during 11.5 seconds). We should raise an error 'Content of the iframe in which the test is currently operating did not load.' in 5 seconds after this. But in this case we didn't raise an error, but were waiting while iframe is reloaded. 
I didn't investigate such problem before so it's hard to say why it happend.

**Comment by georgiy-abbasov** [comment]
After some investigation of `e.isFakeIEEvent` we came to a conclusion, that we don't need this check generally, because next [check](https://github.com/DevExpress/testcafe/blob/master/src/client/core/page-unload-barrier.js#L55) is enough for IE browsers.

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
@georgiy-abbasov So, what about a test for this?

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
FPR
\cc @VasilyStrelyaev, check please

File: test/client/fixtures/core/page-unload-barrier-test.js [hunk 1]
```diff
@@ -101,4 +102,52 @@ $(document).ready(function () {
...
+
+            link.click();
+        });
+
+        asyncTest('Should not resolve waiting promise if page is unloading after click on a link with javascript (GH-608)', function () {
+            var iframeDocument            = iframe.contentDocument;
+            var link                      = iframeDocument.createElement('a');
+            var pageUnloadBarrierResolved = false;
+
+            link.href        = 'javascript:void(window.location.href = "http://example.com")';
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/core/page-unload-barrier-test.js]: This test is redundant. Existing tests cover this scenario

File: test/client/fixtures/core/page-unload-barrier-test.js [hunk 2]
```diff
@@ -101,4 +102,52 @@ $(document).ready(function () {
                 start();
             });
     });
+
+    module('Regression');
+
+    if (browserUtils.isIE9 || browserUtils.isIE10) {
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/core/page-unload-barrier-test.js]: this test should work in all browsers, not only in IE9 and IE10. Just delete this condition

File: test/client/fixtures/core/page-unload-barrier-test.js [hunk 3]
```diff
@@ -101,4 +102,52 @@ $(document).ready(function () {
...
+        asyncTest("Should resolve waiting promise if click on a link with javascript, which doesn't lead to unload (GH-608)", function () {
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/core/page-unload-barrier-test.js]: I see some unnecessary words here. Should be something like 'Should resolve promise if click on a link doesn't lead to page unload (GH-608)'
/cc @VasilyStrelyaev

Comment (VasilyStrelyaev) [inline test/client/fixtures/core/page-unload-barrier-test.js]:
```
        asyncTest("Should resolve the waiting promise when clicking a link with javascript that doesn't lead to unload (GH-608)", function () {
```

File: test/client/fixtures/core/page-unload-barrier-test.js [hunk 4]
```diff
@@ -101,4 +102,52 @@ $(document).ready(function () {
...
+                .wait()
+                .then(function () {
+                    ok(true, 'just check promise is resolved');
+                    start();
+                });
+
+            link.click();
+        });
+
+        asyncTest('Should not resolve waiting promise if page is unloading after click on a link with javascript (GH-608)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/core/page-unload-barrier-test.js]: If it's going to be removed I won't check it...

**Comment by VasilyStrelyaev** [comment]
\r-

File: test/client/fixtures/core/page-unload-barrier-test.js (hunk 3 above)
Comment (VasilyStrelyaev) [inline test/client/fixtures/core/page-unload-barrier-test.js]:
```
        asyncTest("Should resolve the promise when clicking a link that doesn't lead to page unload (GH-608)", function () {
```

File: test/client/fixtures/core/page-unload-barrier-test.js (hunk 2 above)
Comment (georgiy-abbasov) [inline test/client/fixtures/core/page-unload-barrier-test.js]: I've got error on IE 11 when tested without this condition locally with Sauce Labs - "Access is denied"
https://assets.saucelabs.com/jobs/ee41e00cebdf41c48f42ddb9ebfaff6f/0037screenshot.png

Comment (AlexanderMoskovkin) [inline test/client/fixtures/core/page-unload-barrier-test.js]: Let's fix this! =)

File: test/client/fixtures/core/page-unload-barrier-test.js [hunk 5]
```diff
@@ -1,6 +1,7 @@
 var hammerhead    = window.getTestCafeModule('hammerhead');
 var Promise       = hammerhead.Promise;
 var iframeSandbox = hammerhead.sandbox.iframe;
+var browserUtils  = hammerhead.utils.browser;
```
Comment (inikulin) [inline test/client/fixtures/core/page-unload-barrier-test.js]: Test

**Comment by georgiy-abbasov** [comment]
FPR

File: test/functional/fixtures/regression/gh-608/test.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+describe('[Regression](GH-608)', function () {
+    it("Shouldn't wait max unloading time if click on a link with javascript, which doesn't lead to unload", function () {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/regression/gh-608/test.js]: Shouldn't wait for the maximum time... Does it mean it should way for some moderate time? I guess it shouldn't wait at all if the js code doesn't lead to unload, right?
```
    it("Shouldn't wait for page unload if a link with javascript that doesn't lead to unload is clicked", function () {
```

Comment (georgiy-abbasov) [inline test/functional/fixtures/regression/gh-608/test.js]: Yes, you're right.

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
FPR

**Pull request merged**